python/plugins/processing/tests/README.md

```diff
@@ -0,0 +1,96 @@
+Algorithm tests
+===============
+
+To test algorithms you can add entries into `testdata/algorithm_tests.yaml`.
+
+This file is structured with [yaml syntax](http://www.yaml.org/start.html).
+
+A basic test appears under the toplevel key `tests` and looks like this:
+
+  - name: centroid
+    algorithm: qgis:polygoncentroids
+    params:
+      - type: vector
+        location: qgs
+        name: polys.shp
+    results:
+      - id: OUTPUT_LAYER
+        type: vector
+        location: proc
+        name: polys_centroid.geojson
+
+How To
+------
+
+To add a new test you can follow these steps:
```

Comment:
This is a little know feature, but now, in the history, you can right click on an entry and select "Create test", and it will create a Python test for the algorithm. The test is created by this method

https://github.com/qgis/QGIS/blob/master/python/plugins/processing/gui/TestTools.py#L43

I guess we could adapt that to produce a yaml text instead, and copy it to the clipboard, so it can then be easily pasted in the yaml file. It would make it easier for users to contribute tests

What do you think?